SConstruct

```diff
@@ -1433,14 +1433,16 @@ if doConfigure :
 			e.Append( CPPFLAGS = '-DIECORE_WITH_ASIO' )
 	else :
 		sys.stderr.write( "WARNING: boost/asio.hpp not found, some functionality will be disabled.\n" )
-		coreSources.remove( "src/IECore/ClientDisplayDriver.cpp" )
-		coreSources.remove( "src/IECore/DisplayDriver.cpp" )
-		coreSources.remove( "src/IECore/ImageDisplayDriver.cpp" )
-		coreSources.remove( "src/IECore/DisplayDriverServer.cpp" )
-		corePythonSources.remove( "src/IECorePython/ClientDisplayDriverBinding.cpp" )
-		corePythonSources.remove( "src/IECorePython/DisplayDriverServerBinding.cpp" )
-		corePythonSources.remove( "src/IECorePython/DisplayDriverBinding.cpp" )
-		corePythonSources.remove( "src/IECorePython/ImageDisplayDriverBinding.cpp" )
+		coreSources.remove( "src/IECoreImage/ClientDisplayDriver.cpp" )
```

Comment:
`coreSources`only  contains `glob.glob( "src/IECore/*.cpp" )`, so this won't be removing anything. How about we just remove the checks and require asio?

Reply:
It seems asio is only required by IECoreImage and IECoreNuke. Should I make it a primary requirement incase we want it for IECore in the future, or leave it to those libs only?

Reply:
Similarly, -`DIECORE_WITH_SIGNALS` doesn't seemed to be used for anything, so maybe I should remove and make that a hard requirement as well?

Reply:
I think probably just add it to the LIBS for IECoreImage and IECoreNuke, but don't worry about trying to compile them with bits missing if it isn't found. Just assume it'll be there basically, since we've already done a general "is boost available?" check earlier on.

Reply:
> Similarly, -DIECORE_WITH_SIGNALS doesn't seemed to be used for anything, so maybe I should remove and make that a hard requirement as well?

Yeah, all this dates way back to when those libraries were first introduced to boost (somewhere in the 1.3x range I think). VFXPlatform version is 1.61 so it's all ancient history...

Reply:
So at this points, freetype will be the only "optional" dependency of IECore. Should I just make that required too?

Reply:
Let's leave freetype as-is, at least until that dependency is moved out from the main lib.

Reply:
Addressed in 94e0725